### DISCUSSION TITLE
Final rect coordinates for draggable in `onDragEnd` event?

### DISCUSSION THREAD

**Discussion by alspdx** [clauderic/dnd-kit#236] [Q&A] [answered]
Hi everyone, 

In our use case for DnD we are moving multiple draggable "ratable" items from an unrated list into a droppable "rating container" component. We need to calculate the item's absolute position in relation to the rating container in the `onDragEnd` event handler. The item's position within the rating container boundaries is used to calculate the item rating and compare ratings with the other items — it isn't relevant after dragging to know where the item came from (`delta` or `transform`, though these are obviously useful for positioning while actively dragging), we just need the final item location data to calculate position. 

The coordinates for the drop container are available in the `DragEndEvent` at `event.over.rect` (`LayoutRect`) but, unless I'm missing something, I'm not seeing the draggable item client layout rect data for `event.active.rect.current` alongside the other details about the active draggable item (both `rect.current.initial` and `rect.current.translated` properties are being set to `null` before `onDragEnd` is triggered). 

Of course I _can_ track the location while the item is actively dragging from `onDragMove` and use the last coordinates in the `onDragEnd` handler to calculate position, but that's another layer of state that would need to be managed. It would be ideal to have all of the same active element location data that's available in the `onDragMove` event accessible in the `onDragEnd` event.

Interestingly `react-dnd` has the opposite problem for us — "drag source" element position data is available in one of the arguments to the drop event handler, but "drop target" location data is not and would need to be handled otherwise.

Are there any plans to make final coordinates available from within the `onDragEnd` event handler? Or am I missing something in the `DragEndEvent`?

Thanks!

**Comment by clauderic** [comment, marked as answer]
Hey @alspdx, sounds like a bug. Mind re-opening this as an issue?
